src/quo/components/tabs/tabs/view.cljs

```diff
@@ -118,87 +119,99 @@
+           scroll-on-press?    false
+           size                default-tab-size}
+    :as   props}]
+  (let [[active-tab-id set-active-tab-id] (rn/use-state default-active)
+        [fading set-fading]               (rn/use-state {:fade-end-percentage fade-end-percentage})
+        flat-list-ref                     (rn/use-ref-atom nil)
+        set-flat-list-ref                 (rn/use-callback #(reset! flat-list-ref %))
+        set-initial-scroll-poisition      (rn/use-callback (fn []
+                                                             (js/setTimeout
+                                                              (fn []
+                                                                (.scrollToIndex
+                                                                 @flat-list-ref
+                                                                 #js
+                                                                  {:animated false
+                                                                   :index
+                                                                   (utils.collection/first-index
+                                                                    #(= active-tab-id (:id %))
+                                                                    data)}))
+                                                              1)))]
+    (if scrollable?
+      [rn/view {:style {:margin-top (- (dec unread-count-offset))}}
+       [masked-view-wrapper
+        {:fade-end-percentage (get fading :fade-end-percentage) :fade-end? fade-end?}
+        [(if in-scroll-view?
+           gesture/flat-list
+           rn/flat-list)
+         (merge
+          (dissoc props
+           :default-active
+           :fade-end-percentage
+           :fade-end?
+           :on-change
+           :scroll-on-press?
+           :size)
+          {:ref set-flat-list-ref
+           :style style
+           ;; The padding-top workaround is needed because on Android
+           ;; {:overflow :visible} doesn't work on components inheriting
+           ;; from ScrollView (e.g. FlatList). There are open issues, here's just one about this
+           ;; topic: Https://github.com/facebook/react-native/issues/31218
+           :content-container-style
+           (assoc container-style :padding-top (dec unread-count-offset))
+           :horizontal true
+           :scroll-event-throttle 64
+           :shows-horizontal-scroll-indicator false
+           :data data
+           :key-fn (comp str :id)
+           :on-scroll-to-index-failed identity
+           :on-scroll (partial on-scroll-handler
+                               {:fade-end-percentage fade-end-percentage
+                                :fade-end?           fade-end?
+                                :fading              fading
+                                :set-fading          set-fading
+                                :on-scroll           on-scroll})
```

Comment:
This looks a bit confusing. We're passing ` fade-end-percentage` and `fading` state, which is just `{:fade-end-percentage  fade-end-percentage}`. Maybe `on-scroll-handler` could be a custom-hook that contains and updates the `fading` state, which would be returned together with the scroll handler e.g.

```clj
(let [...
    ;; `fading` state would be inside the hook instead of passing both the state and the setter to
    ;; `on-scroll-handler`, as it seems like no other place updates it
    {:keys [updated-fade-end-percentage on-scroll-handler]
         (use-tab-scroll {:fade-end-percentage fade-end-percentage
                          ...})}]
...)
```